django/db/models/deletion.py

```diff
@@ -269,6 +272,14 @@ def delete(self):
         # number of objects deleted for each model label
         deleted_counter = Counter()
 
+        # Optimize for the case with a single obj and no dependencies
+        if len(self.data) == 1 and len(instances) == 1:
+            instance = list(instances)[0]
+            if self.can_fast_delete(instance):
+                with transaction.mark_for_rollback_on_error():
```

Comment:
Looks like this should used `self.using`, I'll submit a PR for it.

Reply:
Well spotted :+1: